apps/web-app/src/pages/main/battle/attack/list.tsx

```diff
@@ -1,30 +1,60 @@
 import DarkThroneClient from '@darkthrone/client-library';
-import { useEffect, useState } from 'react';
+import { useCallback } from 'react';
 import { Avatar } from '@darkthrone/react-components';
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useSearchParams } from 'react-router-dom';
 import SubNavigation from '../../../../components/layout/subNavigation';
 import { PlayerObject } from '@darkthrone/interfaces';
+import { Paginator } from '../../../../libs/pagination';
+import Pagination from '../../../../components/pagination';
 
 interface AttackListPageProps {
   client: DarkThroneClient;
 }
 export default function AttackListPage(props: AttackListPageProps) {
   const navigate = useNavigate();
-  const [players, setPlayers] = useState<PlayerObject[]>([]);
 
-  useEffect(() => {
-    const fetchPlayers = async () => {
-      const playersFetch = await props.client.players.fetchAllPlayers();
-      if (playersFetch.status === 'fail') {
-        console.error(playersFetch.data);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const pageParam = searchParams.get('page');
+  const currentPageNumber = pageParam ? Math.max(1, parseInt(pageParam)) : 1;
+  const pageSize = 100;
+
+  const fetchForPage = useCallback(
+    async (pageNumber: number) => {
+      const fetched = await props.client.players.fetchAllPlayers(
+        pageNumber,
+        pageSize,
+      );
+
+      if (fetched.status === 'fail') {
         return;
       }
-      setPlayers(
-        playersFetch.data.sort((a, b) => a.overallRank - b.overallRank),
-      );
-    };
-    fetchPlayers();
-  }, [props.client.players]);
+
+      return {
+        items: fetched.data.items,
+        meta: fetched.data.meta,
+      };
+    },
+    [props.client.players],
+  );
+
+  const { setPage, totalItems, data } = Paginator<PlayerObject>({
```

Comment:
```suggestion
  const { setPage, totalItems, data: players } = Paginator<PlayerObject>({
```
Nit:

I prefer using very suggestive names when possible, making it easier to understand what things are when reading variable names instead of checking where the variable is given a value.

Reply:
I understand what you are saying here. However, this is a generic interface and so the variable itself is what is it. We could declare an alternate name perhaps like `data as players`. Please do note though that the correct typings are passed. Typescript (and VSCode) is fully aware that data contains players and the intellisence works.

Reply:
Sounds good, feel free to add "as players" if you want to, but I'll resolve this.

Thanks for explaining.